clwb/tests/integrationtests/com/google/idea/blaze/clwb/base/ClwbIntegrationTestCase.java

```diff
@@ -273,27 +273,42 @@ protected boolean runQuerySync() {
     return false;
   }
 
-  protected boolean enableAnalysisFor(VirtualFile file) throws ExecutionException {
+  protected SyncOutput enableAnalysisFor(VirtualFile file) throws ExecutionException {
+    final var context = BlazeContext.create();
+
+    final var output = new SyncOutput();
+    output.install(context);
+
     final var manager = QuerySyncManager.getInstance(myProject);
     final var targets = manager.getTargetsToBuild(file).targets();
 
-    final var future = manager.enableAnalysis(
-        targets,
-        QuerySyncActionStatsScope.createForFile(getClass(), null, file),
-        TaskOrigin.USER_ACTION
-    );
+    final var projects = manager.getLoadedProject();
+    assertThat(projects).isPresent();
+
+    final var future = CompletableFuture.runAsync(() -> {
+      try {
+        projects.get().enableAnalysis(context, targets);
+      } catch (Exception e) {
+        LOG.error("enable analysis failed", e);
+      }
+    }, ApplicationManager.getApplication()::executeOnPooledThread);
 
     while (!future.isDone()) {
       PlatformTestUtil.dispatchAllInvocationEventsInIdeEventQueue();
     }
 
+    context.close();
+    LOG.info(String.format("PROJECT BUILD LOG:%n%s", output.collectLog()));
+
     try {
-      return future.get();
+      future.get();
+    } catch (ExecutionException e) {
+      LOG.error("enable analysis failed", e);
     } catch (InterruptedException e) {
-      fail("enable analysis was interrupted");
+      LOG.error("enable analysis was interrupted", e);
```

Comment:
Why are the exceptions swallowed here?

Reply:
They are not, log error throws in tests and provides more details then `fail`.